docs/adrs/007-minifront-v2-architecture.md

```diff
@@ -0,0 +1,261 @@
+
+The original minifront established key principles:
+
+- **Privacy preserving**: Maximally client-side to protect user privacy
+- **Deployable by PD nodes**: Statically rendered without special server configuration
+- **Hash routing**: Enabling deployment on generic static file servers
+
+Minifront-v2 builds upon these foundations while addressing scalability, maintainability, and developer experience challenges encountered in the original implementation.
+
+## Decision
+
+### Preserved Core Principles
+
+Minifront-v2 maintains **complete parity** with the original architecture's fundamental privacy and deployment characteristics:
+
+1. **Client-side rendering**: Fully client-side React application with no server-side dependencies
+2. **Hash routing**: Continues using React Router's hash router for deployment flexibility
+3. **Static build output**: Single `index.html` deployable on any static file server
+4. **Privacy-first**: No code-splitting by route to prevent network traffic analysis
+5. **PD node compatibility**: Can be served directly by Penumbra daemon nodes
+
+### Architectural Evolution
+
+Minifront-v2 introduces several significant architectural improvements:
+
+#### 1. Feature-Sliced Design (FSD) Architecture
+
+```
+src/
+├── app/           # Application initialization & routing
+├── pages/         # Route-level page components
+├── widgets/       # Complex standalone UI widgets
+├── features/      # Business logic features
+├── entities/      # Business entities & domain logic
+└── shared/        # Shared utilities, services, stores
+```
+
+**Benefits:**
+
+- Clear separation of concerns by abstraction level
+- Improved scalability for larger feature sets
+- Better code organization and discoverability
+- Reduced coupling between components
+
+#### 2. MobX State Management
+
+**Divergence from original**: Replaces Zustand + TanStack Query with MobX stores
+
+```typescript
+// Original pattern (Zustand slice)
+export const useBalancesStore = create<BalancesSlice>((set, get) => ({
+  balances: [],
+  loading: false,
+  // ... manual state management
+}));
+
+// v2 pattern (MobX store)
+export class BalancesStore {
+  @observable balances: Balance[] = [];
+  @observable loading = false;
+
+  @computed get balancesByAccount() {
+    // Automatically memoized computed values
+  }
+}
+```
+
+**Benefits:**
+
+- **Performance**: Granular reactivity - only components observing changed data re-render
+- **Developer Experience**: Direct state mutations with automatic change detection
+- **Type Safety**: Full TypeScript inference without boilerplate
+- **Computed Values**: Memoized derivations that update only when dependencies change
+
+#### 3. Service Layer Architecture
+
+**New addition**: Centralized service layer for business logic separation
+
+```typescript
+// Services handle data operations
+export class PenumbraService {
```

Comment:
though: this pattern actually contradicts the FSD, which is all about decoupling and modules. I imagine this class definition growing to 500+ lines of code with time.

suggestion: keep the requests logic in the `api` directory of a module you're implementing. If the logic gets shared, move it 1 layer down the FSD. For example, if `useBalances` is defined in a Swap page, when you need to reuse it between pages - you refactor it by moving to `entities` or `features`.